### PR TITLE
Stop/restart VPN when bad health is detected in internal builds

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1201629177365358/f
* https://app.asana.com/0/488551667048375/1201641519620933/f
* https://app.asana.com/0/488551667048375/1201641519620969/f
* https://app.asana.com/0/488551667048375/1201653659122893/f

### Description
As part of the effort to understand bad health conditions in the VPN and test mitigataions. This PR adds the functionality of automatically restarting the VPN when bad health is detected.

* The automatic restart only happens in internal builds
* We add an internal feature to enable/disable bad health monitoring

In addition this PR:
* Fixed bug that make some health rates to go above 100%
* Adjusts the sensitivity of some alerts
* Streamlines the packet processing in the UDP pipeline
* Splits TCP/UDP health metrics

Incidentally this PR fixes the bug described in [here](https://app.asana.com/0/488551667048375/1201653659122893/f). Edge case around cohort calculation.

### Steps to test this PR

_Test Internal builds_
- [x] fresh install internal build
- [x] open app and enable AppTP
- [x] go to settings -> appTP settings -> diagnostics
- [x] tap on `Bad Health` CTA
- [x] verify `Internal flavor detected, restarting the VPN...` message shows in the logcat (can take up to 30s)
- [x] verify `m_atp_health_monitor_report` pixel is sent
- [x] verify `m_atp_health_alert_tunReadAlerts_d` is sent
- [x] verify the next `Internal flavor detected, restarting the VPN...` message shows at least 1min after the one before
- [x] verify **only** `m_atp_health_monitor_report` pixel is sent and NOT `m_atp_health_alert_tunReadAlerts_d`
- [x] go to settings -> appTP settings -> diagnostics
- [x] tap on `Good health` CTA
- [x] verify `No alerts, clearing storage` shows in the logcat (can take up to 30s)

_Test bad health information storing/removal_
- [x] install internal build
- [x] open app and enable AppTP
- [x] go to settings -> appTP settings -> diagnostics
- [x] tap on `Bad Health` CTA
- [x] go to "App Tracking protection" screen and report breakage for any app
- [x] verify the `breakageMetadata` in the `m_atp_breakage_report` pixel contains the bad health state
- [x] go to settings -> appTP settings -> diagnostics
- [x] tap on `Good health` CTA
- [x] go to "App Tracking protection" screen and report breakage for any app
- [x] verify the `breakageMetadata` in the `m_atp_breakage_report` pixel DOES NOT contain the bad health state

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #1665**  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: vpn/src/main/java/com/duckduckgo/mobile/android/vpn/bugreport/AppHealthStateCollector.kt [hunk 1]
```diff
@@ -32,7 +32,7 @@ class AppHealthStateCollector @Inject constructor(
     override val collectorName: String = "latestAppBadHealth"
 
     override suspend fun collectVpnRelatedState(appPackageId: String?): JSONObject {
-        val healthState = appHealthDatabase.appHealthDao().latestHealthState()
+        val healthState = appHealthDatabase.appHealthDao().remove()
```
Comment (malmstein) [inline vpn/src/main/java/com/duckduckgo/mobile/android/vpn/bugreport/AppHealthStateCollector.kt:35]: maybe I miss the point, but if you remove something is it meant to return a value?

**Review by malmstein — APPROVED** [review]
very nice work @aitorvs . just a couple comments about naming, everything works as expected... when Gradle wants to...

File: vpn/src/main/java/com/duckduckgo/mobile/android/vpn/bugreport/AppHealthStateCollector.kt (hunk 1 above)
Comment (aitorvs) [reply vpn/src/main/java/com/duckduckgo/mobile/android/vpn/bugreport/AppHealthStateCollector.kt:35]: I took the same approach as what eg. `LRUCache` API. Where remove removes from the cache and gives you the removed item 🤷
